falcon/middleware.py

```diff
@@ -42,38 +45,53 @@ class CORSMiddleware(object):
 
     def __init__(
         self,
-        allow_origins: Union[str, Iterable[str]] = '*',
+        allow_origins: Union[str, Iterable[str], OriginFilter] = '*',
         expose_headers: Optional[Union[str, Iterable[str]]] = None,
-        allow_credentials: Optional[Union[str, Iterable[str]]] = None,
+        allow_credentials: Optional[Union[str, Iterable[str], OriginFilter]] = None,
     ):
-        if allow_origins == '*':
-            self.allow_origins = allow_origins
+        self.allow_origins_wildcard = False
```

Comment:
Style: prefix `allow_origins_wildcard` name with an underscore; I don't think there is any need to make this part of the public interface.